### PR TITLE
Mapeditor improvements

### LINKED ISSUE #2590: Map validation unusable
**Describe the bug**
Cannot see the information from map validation

**Screenshots**
image

**Version**
 - OS: [e.g. Windows, 
 - Version: [beta

### LINKED ISSUE #3087: add vmap extension
**Describe the bug**
If you save map in map editor and don't add the ".vmap" extension by yourself it's missing. File without extension is not shown in VCMI.

**Expected behavior**
Map editor should automatically add ".vmap" is missing.

Also it would be grerat if the "Maps" folder could be the default one in the saving dialog.

**Version**
 - OS: Ubuntu
 - Version: develop

### LINKED ISSUE #3864: Add a search box in mapeditor when setting army
**Describe your proposal**
When setting army in heroes or towns, it is inconvenient for user to select a creature which he want to add. 

屏幕截图 2024-05-01 002950

So much creatures, hard to pick.

### PR DESCRIPTION
**Pull request by Laserlicht**
Fixes #3087 (was part of stalled PR #3115)
Fixes #3864
Fixes #2590

### CODE REVIEW

**Review by kambala-decapitator — APPROVED** [review]

File: mapeditor/main.cpp [hunk 1]
```diff
@@ -13,6 +13,10 @@
 
 int main(int argc, char * argv[])
 {
+#if QT_VERSION < QT_VERSION_CHECK(6, 0, 0)
+	QApplication::setAttribute(Qt::AA_EnableHighDpiScaling);
+#endif
+
```
Comment (IvanSavenko) [inline mapeditor/main.cpp:19]: Not against it, but why we only do this for Qt 5?

Comment (kambala-decapitator) [reply mapeditor/main.cpp:19]: in Qt 6 it's a warning and CI would fail due to -Werror

Comment (IvanSavenko) [reply mapeditor/main.cpp:19]: Deprecation warning?

Comment (Laserlicht) [reply mapeditor/main.cpp:19]: This option is default in QT6 and cannot disabled.

**Pull request merged**